labrad/crypto.py

```diff
@@ -0,0 +1,130 @@
+import base64
+import hashlib
+import logging
+import os
+
+
+try:
+    from twisted.internet import ssl
+    TLS = True
+except ImportError:
+    logging.warning('pyOpenSSL not found. Without encryption you will only be '
+                    'able to connect to the labrad manager on localhost.')
+    TLS = False
+
+
+certs_path = os.path.expanduser('~/.labrad/client/certs')
+# Location where we store trusted self-signed manager certs.
+# The manager cert file for a given hostname is named <hostname>.cert
+# Note that real certs signed by public CAs are not stored here,
+# because we can validate them with the system's trusted certificates.
+
+
+def tls_options(hostname, cert_string=None):
+    """Get certificate options for TLS connections.
+
+    If we are unable to use TLS because pyOpenSSL is not installed, this will
+    raise an exception.
+
+    Args:
+        hostname (string): The hostname we are trying to connect to with TLS.
+        cert_string (string): A certificate in PEM format. If provided, we
+            return CertificateOptions that will trust only the given cert
+            (assuming its canonical name attribute matches the given hostname).
+            If not provided, we will try to load a suitable certificate for
+            this hostname from our store of trusted self-signed certificates,
+            and otherwise fall back to just using the trust roots configured
+            for the OS to validate the server certificate.
+
+    Returns (twisted.internet.ssl.CertificateOptions):
+        Certificate suitable for use by TLS clients to connect to the given
+    """
+    if not TLS:
+        raise Exception('Unable to connect to labrad with TLS. Please install pyOpenSSL.')
```

Comment:
style nit: It's customary to add a "Raises" section to the docstring if you expliclty raise exceptions in the code